benchmark/metrics/cputime.py

```diff
@@ -0,0 +1,216 @@
+            # ignore blank lines
+            return
+        try:
+            name, formatted_cputime = line.split()
+            parts = formatted_cputime.split(b'-', 1)
+            # The last item is always HH:MM:SS. Split it and convert to
+            # integers.
+            h, m, s = [int(t) for t in parts.pop().split(b':')]
+            # Anything left is the number of days
+            days = int(parts[0]) if parts else 0
+            cputime = ((days * 24 + h) * 60 + m) * 60 + s
+        except ValueError as e:
+            e.args = e.args + (line,)
+            raise e
+
+        self.result[name] = self.result.get(name, 0) + cputime
+
+
+class SSHRunner:
+    """
+    Run a command using ssh.
+
+    :ivar reactor: Twisted Reactor.
+    :ivar user: Remote user name.
+    """
+
+    def __init__(self, reactor, user=b'root'):
+        self.reactor = reactor
+        self.user = user
+
+    def run(self, node, command_args, handle_stdout):
+        """
+        Run a command using SSH.
+
+        :param Node node: Node to run command on.
+        :param [str] command_args: List of command line arguments.
+        :param callable handle_stdout: Function to handle each line of output.
+        :return: Deferred, firing when complete.
+        """
+        d = run_ssh(
+            self.reactor,
+            self.user,
+            node.public_address.exploded,
+            command_args,
+            handle_stdout=handle_stdout,
+        )
+        return d
+
+
+def get_node_cpu_times(runner, node, processes):
+    """
+    Get the CPU times for processes running on a node.
+
+    :param runner: A method of running a command on a node.
+    :param node: A node to run the command on.
+    :param processes: An iterator of process names to monitor.
+    :return: Deferred firing with a dictionary mapping process names to
+        elapsed cpu time.  If an error occurs, returns None (after
+        logging error).
+    """
+    parser = CPUParser()
+    d = runner.run(
+        node,
+        _GET_CPUTIME_COMMAND + [b",".join(processes)],
+        handle_stdout=parser.lineReceived,
+    )
+    d.addCallback(lambda ignored: parser.result)
+    d.addErrback(eliot.writeFailure)
+    return d
+
+
+def get_cluster_cpu_times(clock, runner, nodes, processes):
+    """
+    Get the CPU times for processes running on a cluster.
+
+    :param clock: Twisted Reactor.
+    :param runner: A method of running a command on a node.
+    :param node: Node to run the command on.
+    :param processes: An iterator of process names to monitor.
+    :return: Deferred firing with a dictionary mapping process names to
+        elapsed cpu time.  If an error occurs, returns None (after
+        logging error).
+    """
+    return gather_deferreds(list(
+        get_node_cpu_times(runner, node, processes)
+        for node in nodes
+    ))
+
+
+def compute_change(labels, before, after):
+    """
+    Compute the difference between times retrieved ast different times.
```

Comment:
This comment could be clearer:
`Compute the difference between CPU times retrieved at different wallclock times.`